src/Commands/Site/GetSite.cs

```diff
@@ -10,13 +10,67 @@ namespace PnP.PowerShell.Commands.Site
     {
+        [Parameter(Mandatory = false)]
+        public SwitchParameter GetVersionPolicy;
+
         protected override void ExecuteCmdlet()
         {
             DefaultRetrievalExpressions = new Expression<Func<Microsoft.SharePoint.Client.Site, object>>[] { s => s.Url, s => s.CompatibilityLevel };
             var site = ClientContext.Site;
-            ClientContext.Load(site, RetrievalExpressions);
-            ClientContext.ExecuteQueryRetry();
-            WriteObject(site);
+
+            if (ParameterSpecified(nameof(GetVersionPolicy)))
+            {
+                site.EnsureProperties(s => s.Url, s => s.VersionPolicyForNewLibrariesTemplate, s => s.VersionPolicyForNewLibrariesTemplate.VersionPolicies);
+
+                var vp = new VersionPolicy();
+                vp.Url = site.Url;
+
+                if (site.VersionPolicyForNewLibrariesTemplate.VersionPolicies.ServerObjectIsNull == true)
+                {
+                    vp.Description = "No Site Level Policy Set";
+                }
+                else
+                {
+                    site.EnsureProperties(s => s.VersionPolicyForNewLibrariesTemplate, s => s.VersionPolicyForNewLibrariesTemplate.MajorVersionLimit, s => s.VersionPolicyForNewLibrariesTemplate.VersionPolicies);
+
+                    vp.DefaultTrimMode = site.VersionPolicyForNewLibrariesTemplate.VersionPolicies.DefaultTrimMode.ToString();
+
+                    if (site.VersionPolicyForNewLibrariesTemplate.VersionPolicies.DefaultTrimMode == VersionPolicyTrimMode.AutoExpiration)
+                    {
+                        vp.Description = "Site has Automatic Policy Set";
+                    }
+                    else
+                    {
+                        if (site.VersionPolicyForNewLibrariesTemplate.VersionPolicies.DefaultTrimMode == VersionPolicyTrimMode.ExpireAfter)
+                        {
+                            vp.DefaultExpireAfterDays = site.VersionPolicyForNewLibrariesTemplate.VersionPolicies.DefaultExpireAfterDays.ToString();
+                            vp.Description = "Site has Manual settings with specific count and time limits";
+                        }
+                        else if (site.VersionPolicyForNewLibrariesTemplate.VersionPolicies.DefaultTrimMode == VersionPolicyTrimMode.NoExpiration)
+                        {
+                            vp.Description = "Site has Manual settings with specific version count limit and no time limits";
+                        }
+                        vp.MajorVersionLimit = site.VersionPolicyForNewLibrariesTemplate.MajorVersionLimit.ToString();
+                    }
+                }
+
+                WriteObject(vp);
+            }
+            else
+            {
+                ClientContext.Load(site, RetrievalExpressions);
+                ClientContext.ExecuteQueryRetry();
+                WriteObject(site);
+            }
         }
     }
+
+    public class VersionPolicy
```

Comment:
Could you put it in its own .cs file under src\Commands\Model\SharePoint instead and add some code comments as to what the class does and each of the properties stand for?

Reply:
Thanks @KoenZomers  for the suggestion! Should I create a new command for my VersionPolicy? Because I saw "[Cmdlet(VerbsCommon.Get, "PnPSiteSensitivityLabel")]" command also get properties from site, it uses a separate command, but not the Get-PnPSite.

Reply:
Yes please. Ideally one cmdlet always return the same type so we can add this at the top:

[OutputType(typeof(Microsoft.SharePoint.Client.Site))]

If it returns anything else, it should have its own cmdlet. Please remember to also write the documentation .md files if you create a new cmdlet.

Reply:
Thanks @KoenZomers for the confirmation! Created a new cmdlet, added md file, reverted the changes in Get-PnPSite command and md files. Also updated the test results in the PR description. Could you please review again? Thanks so much!

Reply:
Hi @KoenZomers , Could you please check the PR again? Thanks a lot!